### DISCUSSION TITLE
👋 你来了！这里是讨论毕业论文模板的地方

### DISCUSSION THREAD

**Discussion by Tr0py** [Tr0py/NKU-thesis-template-2020#18] [Announcements]
你可以在这里提出：
* 使用模板**遇到的问题**
* 对于模板的**改进想法**

对于大佬们：如果有时间和能力，可以
* 发现明确问题可以直接提出[issue](https://github.com/Tr0py/NKU-thesis-template-2020/issues)。
* 拥有修复方法可以直接提出[PR](https://github.com/Tr0py/NKU-thesis-template-2020/pulls)。

**Comment by YuanzhuZhan** [comment]
求助，使用overleaf，添加校外指导教师后编译并不显示
![Uploading issue.png…]()

**Reply by YuanzhuZhan** [reply]
![issue](https://github.com/Tr0py/NKU-thesis-template-2020/assets/96646670/92c32ef8-4b5a-43c5-9056-53fc84b0c0d3)

**Reply by Tr0py** [reply]
感谢反馈。修复方法是将 `NKThesis.sty` 836行改为

```latex
    \NKT@tp@item{\hskip 5em}{noneval}{xiaowaidaoshi}{noneval}{noneval}\\
```

<img width="729" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/2e9d3574-eef3-4339-824d-ab6d7d23d939">

之后的效果：

<img width="390" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/6453b755-a7eb-4738-8875-cd3d40eb543f">


如果需要显示“校外导师” 可以把 `\hskip 5em` 改成对应的文字。

**Reply by YuanzhuZhan** [reply]
已解决，十分感谢！

**Comment by wyhzhen6** [comment]
想问一下模板应该怎么放伪代码呢，提示LaTeX Error: Environment breakablealgorithm undefined

**Reply by Tr0py** [reply]
是否用了algorithmcx 包？

试着删掉
```
\usepackage{algorithmicx}
```
然后加上
```
\usepackage{algorithm}% http://ctan.org/pkg/algorithms
\usepackage{algpseudocode}% http://ctan.org/pkg/algorithmicx
```

**Reply by wyhzhen6** [reply]
👌解决了，十分感谢！

**Comment by STEVENTAN100** [comment]
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93769514/9f6d1bdd-2242-4229-a2ff-330d38c433db)
求助：封面论文题目太长溢出下划线边界请问如何处理？如何把下划线长度延长？

**Reply by STEVENTAN100** [reply]
英文题目我可以挪到第二行显示，但是中文题目多出的两个字挪到第二行很别扭，希望能有解决办法！

**Reply by Tr0py** [reply]
所以需求是把中文题目这一行的下划线延长吗？

**Reply by STEVENTAN100** [reply]
是的，把中文下划线延长，或者超出边界能自适应换行都可以（最好第二行多几个字，不然很别扭），谢谢啦！

**Reply by Tr0py** [reply]
在`NKThesis.sty`中把

```
\def\NKT@tp@item#1#2#3#4#5{%
  #1%
  \hskip 0.5em%
  \underline{\hbox to 53mm{\zihaosi\NKT@keyvalue{#2} \zihaosi\kaiti\NKT@keyvalue{#3}
      %
      \zihaosi\songti\NKT@keyvalue{#4}
      %
      \zihaosi\kaiti\NKT@keyvalue{#5}\hfil}}}
  ```

53mm 改成更大的值，如73mm，以此增加所有下划线长度。

<img width="416" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/6a3bb64b-58e4-48d7-9984-7d28daa19d34">

<img width="644" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/a72fe57d-f92c-4eee-a6c3-77ed4d73dc06">

**Reply by STEVENTAN100** [reply]
好的，非常感谢！中外文题目下划线应该也可以更改吧？

**Reply by Tr0py** [reply]
喔你说的是题目。题目的话用同样的方式修改 `\def\NKT@tb@item#1#2` 中的值
```
\def\NKT@tb@item#1#2{%
  #1%
  \hskip 0.5em%
  \underline{\hbox to 115mm{\zihaoxiaosan{\cukai\NKT@keyvalue{#2}\hfil}}}}
```

把115mm 改成更短的值，让你第二行多一些字。或者延长，把题目都放第一行。

<img width="416" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/9c7032e1-05f1-4408-99cb-ff9802fc8571">

**Comment by STEVENTAN100** [comment]
我还想求助一个问题：附录的模板请问可以在未来版本中更新吗？

**Reply by Tr0py** [reply]
对于附录模板有什么需求呢？可以提出来issue, 应该会更新。

**Comment by Kemoralocy** [comment]
求助，使用overleaf，在使用数学环境下\boldsymbol未加粗（\mathbf正常加粗）
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/161679476/4f65e2e4-8aad-45c8-8e74-7a1564e8387b)

**Reply by Tr0py** [reply]
使用 `\usepackage{bm}` 了吗？

```latex
$$\boldsymbol{XY}XY\mathbf{XY}$$
```

<img width="147" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/7a28910b-0ae5-4ada-81e5-843a98ca9f75">

**Reply by Kemoralocy** [reply]
解决了，谢谢大佬！

**Comment by STEVENTAN100** [comment]
求助，我发现模板中章和节的间距过大，以及图片和下一段之间的间距过大，我觉得这样不够紧凑，请问应该如何调整？
图一
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93769514/83e6d8c5-3402-4e9d-b8fd-ca416c9a5cb3)
图二
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93769514/e5aaf4b6-5e22-423d-8f43-6cbdb19010eb)

**Reply by Tr0py** [reply]
1. 调整章和节的间距：修改 `NKThesis.sty` 中的 `\def\belowchapterspace{18pt}`  至想要的值（可以为负数）
2. 图片和下一段之间的间距：推荐使用`\vspace{}` 对每个图片进行修改： 如`\vspace{-10mm}`

**Reply by STEVENTAN100** [reply]
好像看起来效果并不明显，算了，公式的行距也很大，但是为了凑够15篇我就不改行距了，谢谢学长😊

**Comment by nkucszzt** [comment]
请问如何更改什么配置可以打印全部的参考文献而不是只有引用过的参考的文献

**Reply by Tr0py** [reply]
如果要显示参考文献请在正文中引用。
参考文献中不应当出现没有引用过的文献。这是latex应有的逻辑，也是写论文的规范。

**Comment by Kemoralocy** [comment]
大佬求助，我使用algorithm和algorithmic包写伪代码，但是最后会出错，
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/161679476/64db2c90-21b3-4030-bcd4-1d1959d71a30)
最后呈现出来基本没问题，但在结尾处会莫名其妙地显示“=0”
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/161679476/2d2e5a2c-0992-4ab1-ac13-5dbe4784d054)
这部分的latex代码为：
`\begin{algorithm}
    \caption{BCD}
    \begin{algorithmic}[1]
        \STATE \textbf{初始化:}\text{选择两组初始点}$(x_1^{-1},x_2^{-1},\cdots,x_s^{-1})=(x_1^0,x_2^0,\cdots,x_s^0).$ 
        \FOR{$k=1,2,\cdots$}
        \FOR{$i=1,2,\cdots$}
        \STATE \text{使用更新公式(1)或 (2) 或 (3) 更新 }$x_i^k$.
        \ENDFOR
        \IF{\text{满足停机条件}}
        \STATE \text{返回}$(x_1^k,x_2^k,\cdots,x_s^k)$,\text{算法终止}.
        \ENDIF
        \ENDFOR   
    \end{algorithmic}
\end{algorithm}
`
这段代码在其他文档环境里是能正常显示的
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/161679476/869ca082-dc3a-42c6-8d69-fd46484be16b)

**Reply by Kemoralocy** [reply]
这部分的latex代码：
\begin{algorithm}
    \caption{BCD}
    \begin{algorithmic}[1]
        \STATE \textbf{初始化:}\text{选择两组初始点}$(x_1^{-1},x_2^{-1},\cdots,x_s^{-1})=(x_1^0,x_2^0,\cdots,x_s^0).$ 
        \FOR{$k=1,2,\cdots$}
        \FOR{$i=1,2,\cdots$}
        \STATE \text{使用更新公式(1)或 (2) 或 (3) 更新 }$x_i^k$.
        \ENDFOR
        \IF{\text{满足停机条件}}
        \STATE \text{返回}$(x_1^k,x_2^k,\cdots,x_s^k)$,\text{算法终止}.
        \ENDIF
        \ENDFOR   
    \end{algorithmic}
\end{algorithm}

**Reply by STEVENTAN100** [reply]
同学，不是全大写的，应该是State，For，EndFor，EndIf，我就说怎么看起来没问题呢😀

**Reply by Kemoralocy** [reply]
你好同学，请问你一般用哪个包来写伪代码呢？
我用algorithm包和algorithmic包的话就得全大写，不全大写反而报错更多了🤦‍♂️
[http://t.csdnimg.cn/Fvgcx](url)
同一段代码在其他latex文档里是能正常输出的╮(╯▽╰)╭

**Reply by STEVENTAN100** [reply]
我自己都写过了，你试试不就知道了吗？不是全大写的

**Reply by STEVENTAN100** [reply]
\begin{algorithm}
\caption{BCD}
\begin{algorithmic}[1]
\State \textbf{初始化:}\text{选择两组初始点}$(x_1^{-1},x_2^{-1},\cdots,x_s^{-1})=(x_1^0,x_2^0,\cdots,x_s^0).$
\For{$k=1,2,\cdots$}
\For{$i=1,2,\cdots$}
\State \text{使用更新公式(1)或 (2) 或 (3) 更新 }$x_i^k$.
\EndFor
\If{\text{满足停机条件}}
\State \text{返回}$(x_1^k,x_2^k,\cdots,x_s^k)$,\text{算法终止}.
\EndIf
\EndFor
\end{algorithmic}
\end{algorithm}

**Reply by Tr0py** [reply]
我使用模板可以正常渲染伪代码：
<img width="1057" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/6f9ee4dd-b2a6-4d38-b6d2-b261cc32060b">

**Reply by Tr0py** [reply]
>  我用algorithm包和algorithmic包的话就得全大写，不全大写反而报错更多了🤦‍♂️

algorithm包的官方语法是首字母大写：https://www.overleaf.com/learn/latex/Algorithms

**Reply by Kemoralocy** [reply]
我明白了，确实是包用的不同
我用的algorithmic包在官方语法里也说了是全大写`https://www.overleaf.com/learn/latex/Algorithms`
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/161679476/5071ff12-578f-4068-9353-8a96c2c683d9)
但模板环境应该是用的algpseudocode包，确实是首字母大写
感谢两位解答，十分感谢！

**Comment by tttran67** [comment]
求助，在公式中加粗时出现伪粗体现象，就是粗体只是原字体的多次复制。例如代码$\boldsymbol{Y}=\{\boldsymbol{y}_1, \ldots, \boldsymbol{y}_n\}$获得的结果是：
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93925012/c05e9055-0261-440f-b2f9-13d1a3a18215)
粗体有很多重影。我在知乎https://www.zhihu.com/question/25290041上看见似乎和使用的字体有关？

**Reply by Tr0py** [reply]
嗯  你可以使用支持粗体数学符号的字体来避免伪粗体

**Reply by Tr0py** [reply]
https://github.com/Tr0py/NKU-thesis-template-2020/issues/65#issuecomment-2062821963

**Reply by Tr0py** [reply]
使用`\usepackage{newtxmath}` 可以解决问题吗？

**Reply by tttran67** [reply]
使用\usepackage{newtxmath} 解决问题了！非常感谢！

**Comment by tttran67** [comment]
求助，请问使用\subsubsection{}后只有文字，没有出现小节编号，目录里也不出现小节该怎么解决呢？例如，代码是\chapter{这是chapter测试}
\section{这是section测试}
\subsection{这是subsection测试}
\subsubsection{这是subsubsection测试}
\paragraph{这是paragraph测试}
\subparagraph{这是subparagraph测试}，编译出的pdf是
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93925012/71e26b48-dd0b-4e56-bc8a-ff37747032ed)
在subsubsection那里没有编号。

**Reply by Tr0py** [reply]
模板并没有对 `subsubsection` 做格式适配。应该是因为在当时的格式要求下没有比subsection更细的分节了。如果还要分节的话可以考虑 `enumerate` .   今年格式要求中有对subsubsection的要求吗？

**Reply by tttran67** [reply]
明白了，所以是为了符合学校的毕设格式要求？今年格式要求中仍然没有比subsection更细的分节。非常感谢！

**Comment by tttran67** [comment]
计网学院在2023年修订的《本科生学位论文排版规范》版本2.1.1中提到：如果关键词一行放不开，第二行要和上一行的“：”后对齐。或许可以考虑增加这一功能？我目前是手动换行，使用\hspace*{}对齐的。例如：
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93925012/8a8ee614-3838-4f88-8bb3-35c3185ac197)

**Reply by Tr0py** [reply]
可以考虑使用`itemize` 包来实现：

```latex
\begin{itemize}
    \item[\textbf{关键词：}] 南开大学，毕业论文，模板，南开大学，毕业论文，模板，南开大学，毕业论文，模板，南开大学，毕业论文，模板
\end{itemize}
```
效果如图：
<img width="651" alt="image" src="https://github.com/Tr0py/NKU-thesis-template-2020/assets/37894237/4363f467-5abb-4cbd-915e-a00882676068">

**Reply by Tr0py** [reply]
这个可以解决问题吗？

**Reply by tttran67** [reply]
貌似已经有关键词模板了，再加一个关键词会重复而且格式不正确？我的代码为：\begin{guanjianci}
\begin{itemize}
    \item[\textbf{关键词：}] 南开大学，毕业论文，模板，南开大学，毕业论文，模板，南开大学，毕业论文，模板，南开大学，毕业论文，模板
\end{itemize}
\end{guanjianci}得到的结果为：
![image](https://github.com/Tr0py/NKU-thesis-template-2020/assets/93925012/fac4971a-c170-4fa9-9bd2-af2380f9379f)

**Reply by Tr0py** [reply]
把 \begin{guanjianci} \end{guanjianci}删掉再用itemize.

**Comment by ZhaoYang0125** [comment]
请问想在模板里添加一些代码但字体一直不对是怎么回事，我使用的`\usepackage{minted}`，代码显示出来字体都是Times New Roman，但我在别的文件中是可以正常显示代码字体的
![屏幕截图 2024-04-27 201942](https://github.com/Tr0py/NKU-thesis-template-2020/assets/106827747/5676a368-e166-41a0-bdbd-93a80abe1589)
![屏幕截图 2024-04-27 201959](https://github.com/Tr0py/NKU-thesis-template-2020/assets/106827747/283f7ce1-d39a-4773-92f6-3e78272b19a6)

**Reply by Tr0py** [reply]
> 但我在别的文件中是可以正常显示代码字体的

别的文件是指？

> 我使用的\usepackage{minted}

能否给出复现问题的代码？

**Reply by ZhaoYang0125** [reply]
找到问题了，因为在模板的格式中定义了
```
\setmonofont[
BoldFont=timesbd.ttf,
ItalicFont=timesi.ttf,
BoldItalicFont=timesbi.ttf,
Path=fonts/
]{times.ttf}
```
将等宽英文字体都设置为了times，而代码字段就是等宽字体，所以无法正常显示代码字体，把这一部分注释掉就好了

**Reply by Tr0py** [reply]
好发现。这段代码有点迷，怎么把monofont定义成了Times New Roman   我去看看。

**Comment by YuanzhuZhan** [comment]
请问下如何在参考文献中给会议论文添加地址。我在.bib文件中添加了location字段，但是编译后并不显示，不知道还要修改哪里

**Reply by Tr0py** [reply]
地址是指什么？想要的显示的样子是什么？

**Reply by YuanzhuZhan** [reply]
就是会议论文开会的地点。比如下面这个。
![issue2](https://github.com/Tr0py/NKU-thesis-template-2020/assets/96646670/7f78949c-2d4f-424d-a411-a7aad64845c8)

**Comment by cliffwalker123** [comment]
请问如果想要删除页眉的话应该怎么做呢？

**Reply by cliffwalker123** [reply]
已经解决
\fancypagestyle{onlyfoot}{
    \renewcommand{\headrulewidth}{0pt}
    \fancyhf{}
    \cfoot{\thepage}
}
NKThesis.sty中第757行：
  % \pagestyle{centerheadings}}
    \pagestyle{onlyfoot}}

**Comment by NagasakiSoyo157** [comment]
求助！毕业论文上报的是英文论文，请问怎么把模板默认语言调为英文？比如第一章，第二章这些默认的标题

**Comment by fscdc** [comment]
我发现最新release一版的模版里面部署到overleaf后我自己的pdf图片无法成功渲染，这个有人解决了吗

**Reply by Tr0py** [reply]
什么报错?

**Reply by fscdc** [reply]
后来解决了，应该是我pdf版本的问题，都转为1.5就行

**Comment by wangyenshu** [comment]
求助!如何将这个模板调整为英文书写的毕业论文?具体就是除了封面，中文摘要外全为英文。
